src/libstore/rewrite-derivation.hh

```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include "derivations.hh"
+#include "store-api.hh"
+#include "types.hh"
+#include "path.hh"
+
+namespace nix {
+
+/**
+ * Replace all occurences of a path in `keys(pathRewrites)` in the derivation
+ * by its associated value.
+ */
+void rewriteDerivation(Store & store, Derivation & drv, const StringMap & pathRewrites);
```

Comment:
I wasn't sure what the best choice was so I left it outside. But sure, I can add it to the class if that seems better to you

Reply:
Mh actually that might be much better because we can make them methods of `BasicDerivation` that are overriden by `Derivation` and remove a fishy `dynamic_cast` in `build.cc`